### ISSUE TITLE
Array methods don't work on readonly arrays

### ISSUE THREAD

**Issue by ark120202** [TypeScriptToLua/TypeScriptToLua#480]
Input:
```ts
const array: ReadonlyArray<string> = ['foo'];
array.forEach(() => {});
```

Current Result:
```lua
local array = {"foo"};
array:forEach(function(____)
end);
```

Expected Result:
```lua
local array = {"foo"};
__TS__ArrayForEach(array, function(____)
end);
```

**Comment by tomblind** [comment]
The more general issue here is ReadonlyArray is not detected as an array type, so all custom operations involving arrays are broken with them.

```ts
const array: ReadonlyArray<string> = ['foo'];
print(array.length); // Doesn't convert .length to # as it should
```
